### PR TITLE
Change project structure to support gradle submodules

### PR DESCRIPTION
**Pull request by ryanc16**
configures the project to have a common module to share code between sub projects

### CODE REVIEW

File: common/src/main/java/com/dudenduke/hudalternatives/common/PlayerMountData.java
Comment (cjh15hub) [inline common/src/main/java/com/dudenduke/hudalternatives/common/PlayerMountData.java:1]: Can you rename the whole package name to common too? I see it is currently in a folder named core

File: deviljars/build.gradle [hunk 1]
```diff
@@ -21,11 +21,15 @@ repositories {
 
     // If you have mod jar dependencies in ./libs, you can declare them as a repository like so.
     // See https://docs.gradle.org/current/userguide/declaring_repositories.html#sub:flat_dir_resolver
-     flatDir {
-         dir '../core/build/libs'
-     }
+    //  flatDir {
```
Comment (cjh15hub) [inline deviljars/build.gradle:23]: Do you want to keep this?

File: common/src/main/java/com/dudenduke/hudalternatives/common/PlayerMountData.java
Comment (cjh15hub) [reply common/src/main/java/com/dudenduke/hudalternatives/common/PlayerMountData.java:1]: Wait, maybe you already did, sorry.

**Comment by cjh15hub** [comment]
Do you think we should remove the MM prefix to some of the files? I had that there so in case I had similar files for devil jars. But now they are pretty separated, and shouldn't be importing the wrong one. MM_Configuration for example.

**Comment by ryanc16** [comment]
> Do you think we should remove the MM prefix to some of the files? I had that there so in case I had similar files for devil jars. But now they are pretty separated, and shouldn't be importing the wrong one. MM_Configuration for example.

I will leave that up to you. I wanted to leave everything mostly in tact from how it was so you could better review and understand the changes made. If you would like to start refactoring class names, that is up to you.

**Pull request merged**